src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RedirectedBasicWriter.cs

```diff
@@ -70,7 +72,39 @@ public override void WriteHtmlAttribute(CSharpRenderingContext context, HtmlAttr
 
         public override void WriteHtmlContent(CSharpRenderingContext context, HtmlContentIRNode node)
         {
-            _previous.WriteHtmlContent(context, node);
+            if (context.Options.DesignTimeMode)
+            {
+                _previous.WriteHtmlContent(context, node);
+                return;
+            }
+
+            const int MaxStringLiteralLength = 1024;
```

Comment:
Either way I'd prefer not doing that as part of this change. In my opinion, that will be a bigger refactor and should be done separately.